### PR TITLE
Fix drop RPC client bug in transaction monitoing protocols, improve cucumber

### PR DESCRIPTION
**Pull request by hansieodendaal**
## Motivation and Context
- Fixed RPC connections being kept open; closing RPC connections in the coinbase monitoring and transaction validation protocols.
- Fixed transaction validation protocol to also validate broadcast transactions.
- Fixed broadcast monitoring protocol to stop when transactions are not valid any more.
- Simplified some cucumber test steps, also with more robust measurements.
- Fixed some failing cucumber test scripts

## How Has This Been Tested?
- RPC connections bugfix tested in PR #2734
- Cucumber tests

## Types of changes

* [X] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [X] I'm merging against the `development` branch.
* [X] I ran `cargo-fmt --all` before pushing.
* [ ] I ran `cargo test` successfully before submitting my PR.
* [X] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by StriderDM — DISMISSED** [review]
LGTM.

Might want to consider removing the commented code in the cucumber tests? Or is that there for future debugging?

**Comment by hansieodendaal** [comment]
> LGTM.
> 
> Might want to consider removing the commented code in the cucumber tests? Or is that there for future debugging?

Hi David, yes, I specifically left it there to make it easier to debug in future, as I do not know of a #Debug directive for Node.js.

File: base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs [hunk 1]
```diff
@@ -562,7 +568,9 @@ where TBackend: TransactionBackend + 'static
             .await?
             .values()
             .filter(|tx| {
-                tx.status == TransactionStatus::MinedUnconfirmed || tx.status == TransactionStatus::MinedConfirmed
+                tx.status == TransactionStatus::Broadcast ||
```
Comment (philipr-za) [inline base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs:571]: Why did you add the Broadcast here? I don't think you can validate a Broadcast transaction because it is not in the chain? If it is in the Broadcast state it will be checked when a broadcast protocol is started for it.

Comment (hansieodendaal) [reply base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs:571]: This is to enable the wallet to update its transaction status after a reorg. It may have broadcasted a transaction using a UTXO that has been reorged out, same as the other two included there. In that case the broadcast monitoring protocol must be started so the transaction can be invalidated.

**Review by SWvheerden — COMMENTED** [review]
It looks like this branch has some other changes in it?
Some of these changes are in PR2775 as well?

**Comment by hansieodendaal** [comment]
> It looks like this branch has some other changes in it?
> Some of these changes are in PR2775 as well?

The only duplicate changes are fixing errors in cucumber scripts I could find so they do not fail in Circle CI

File: base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs (hunk 1 above)
Comment (philipr-za) [reply base_layer/wallet/src/transaction_service/protocols/transaction_validation_protocol.rs:571]: But this will invalidate normal transactions that have been broadcast because they are not in the chain yet. You cannot validate a transaction that is not mined.

**Comment by hansieodendaal** [comment]
Thanks for the comments, let us close this one.
